### PR TITLE
ENH: add dedup method

### PR DESCRIPTION
**Pull request by Hank0626**
## What do these changes do?
- add exact hash
- refine code structure

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

**Review by aresnow1 — DISMISSED** [review]
LGTM

File: python/xorbits/experimental/dedup.py [hunk 1]
```diff
@@ -407,11 +178,8 @@ def __call__(self, df: pd.DataFrame):
 def dedup(
     df: pd.DataFrame,
     col: str,
-    threshold: float = 0.7,
-    num_perm: int = 128,
-    min_length: int = 5,
-    ngram: int = 5,
-    seed: int = 42,
+    method: str = "minhash",
```
Comment (aresnow1) [inline python/xorbits/experimental/dedup.py:181]: Docstring should be changes.

File: python/xorbits/experimental/dedup.py [hunk 2]
```diff
@@ -478,55 +246,78 @@ def dedup(
...
+            num_perm=num_perm,
+            hashranges=HASH_RANGES,
+            ngram_size=ngrams,
+            min_length=min_length,
+            permutations=PERMUTATIONS,
+        )
+
+        op = DataFrameDedup(func=func)
+
+        return from_mars(op(to_mars(df)))
```
Comment (aresnow1) [inline python/xorbits/experimental/dedup.py:323]: Raise NotImplementedError for not supported methods.

File: python/xorbits/experimental/dedup.py (hunk 1 above)
Comment (Hank0626) [reply python/xorbits/experimental/dedup.py:181]: done

File: python/xorbits/experimental/dedup.py (hunk 2 above)
Comment (Hank0626) [reply python/xorbits/experimental/dedup.py:332]: I check the method in https://github.com/xorbitsai/xorbits/blob/4c6b572e0c4fec65190f14be3fa13bbcd8223a95/python/xorbits/experimental/dedup.py#L258

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by codingl2k1 — APPROVED** [review]
LGTM

**Pull request merged**